commands/start.js

```diff
@@ -23,32 +26,55 @@ function main(urc) {
   logger.log(
     `Starting underreact in ${urc.mode} mode. ${chalk.yellow('Wait ...')}`
   );
-  return (
-    del(urc.outputDirectory, { force: true })
-      // webpack needs to run first as others depend on the assets
-      .then(() => new Promise(resolve => watchWebpack(urc, resolve)))
-      .then(() => {
-        Promise.all([
-          writeHtml(urc),
-          writeCss(urc),
-          autoCopy.copy({
-            sourceDir: urc.publicDirectory,
-            destDir: urc.outputDirectory
-          })
-        ]);
-      })
-      .then(() => startServer(urc))
-      .then(() => {
-        watchCss(urc);
-        watchPublicDir(urc);
-      })
+
+  return del(urc.outputDirectory, { force: true }).then(() =>
+    watch(urc, () => startServer(urc))
   );
 }
 
-function watchWebpack(urc, onFirstRun) {
+function watch(urc, onFirstWrite) {
+  let cssOutput;
+  let webpackAssets;
+  let isFirst = true;
+  const htmlCachedWrite = writeHtml(urc);
+  const hasStylesheets = urc.stylesheets.length > 0;
+  const htmlWriter = () => {
+    if (isFirst) {
+      onFirstWrite();
+      isFirst = false;
+    }
```

Comment:
It's unclear to me why an `htmlWriter` function has the responsibility of calling the `onFirstWrite` callback.

Reply:
I think I did that to make sure the server is only started, once we have an HTML on disk. Do you think some other part of code can make signal the server start?